File: app/src/main/java/com/example/mrakopediareader/api/parser/Parser.java
package com.example.mrakopediareader.api.parser;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public interface Parser<T> {
    T fromJsonObject(JSONObject object) throws JSONException;

    ArrayList<T> fromJsonArray(JSONArray array) throws JSONException;
}

File: app/src/main/java/com/example/mrakopediareader/api/parser/Parser.kt
package com.example.mrakopediareader.api.parser

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

interface Parser<T> {
    @Throws(JSONException::class)
    fun fromJsonObject(jsonObject: JSONObject): T

    @Throws(JSONException::class)
    fun fromJsonArray(jsonArray: JSONArray): List<T>
}